Sources/Filters/General/ClipClosedSurface/ccsEdgeLocator.js

```diff
@@ -0,0 +1,30 @@
+export default class CCSEdgeLocator {
+  constructor() {
+    this._edgeMap = new Map();
+  }
+
+  initialize() {
+    this._edgeMap.clear();
+  }
+
+  insertUniqueEdge(i0, i1, i) {
+    // Ensure consistent ordering of edge
+    if (i1 < i0) {
+      // eslint-disable-next-line no-param-reassign
+      [i0, i1] = [i1, i0];
+    }
+
+    // Generate key, which is unique, since we order the edges
+    const key = `${i0}-${i1}`;
+    let node = this._edgeMap.get(key);
+
+    if (!node) {
+      // Didn't find key, so add a new edge entry
+      node = { ptId0: i0, ptId1: i1, edgeId: -1, next: null };
+      this._edgeMap.set(key, node);
+      return [node, i];
+    }
+
+    return [null, node.edgeId];
```

Comment:
if you return `node.edgeId` (i.e. a copy of the edgeId value), you won't be able to modify the `edgeId` of the node outside this function.  

You still must return `node`.